Contribution guide update (#3643)
timescale/docs: contributing.md

# Contribute to Timescale documentation

> Timescale documentation is open for contribution from all community members. The current documentation source is in this repository. The previous source is in the deprecated repository called [docs.timescale.com-content][legacy-source].

Reviewer: Add a section at the bottom with docs for deprecated products and put the old docs link there please.

> This document explains the process and guidelines to follow when contributing.

Reviewer: ```suggestion
This page explains the process and guidelines to follow when contributing to the Timescale documentation.
```

## Contribution process

You can contribute to Timescale documentation in the following ways:

- [Create an issue][docs-issues] in this repository and describe the proposed change. Our doc team will take care of it. 
- [Fork this repository][github-fork] and propose changes in the fork.

> - If you have a write access to this repository, create a branch from `latest` and raise a pull request directly. 

Reviewer: Can you make the first words [Already have write access?] create bla bla

Not sure about having these as sentences without : after the link. your call.

> When raising a PR, you will be prompted to sign a Contributor License Agreement (CLA). This helps to ensure that the community is free to use your contributions.

Reviewer: present tense everywhere please.

> The documentation site is statically generated with [Gatsby][gatsby]. Its source code is in a separate private 

Reviewer: The documentation site in a separate private repository using [Gatsby] ...

This section has the procedure backwards. Can you make the whole thing into a procedure please.
1. Get the source code:
    * Fork
    * already have access:
2. Whatever

repository, which pulls in the content from this repository on each build.

Once you raise a PR for any branch, GitHub will **automatically** generate a preview for your changes and attach the link in the comments. Any new commits will be visible at the same URL. If you don't see the latest changes, try an incognito browser window.

> ## Documentation guidelines

Reviewer: Can you remove this heading and move everything up one level please

This section provides pointers on how to write, structure, and organize your contribution. Do not worry if you don't get it right on the first try - our documentation team is always there to help.

### Language

Aim to write in a clear, concise, and actionable manner. Timescale documentation uses the [Google Developer Documentation Style Guide][google-style] with the following exceptions:

- Do not capitalize the first word after a colon.
- Use code font (back ticks) for UI elements instead of bold.

> ### Documentation structure

Reviewer: Can you remove this heading and move everything up one level please,

    |`errors`| object of form `{language: string, message: string}` |No| The error, if any, related to the troubleshooting entry. Displayed as a code block right underneath the title. `language` is the programming language to use for syntax highlighting. |
    |`keywords`| array of strings                                     |No| These are displayed at the bottom of every troubleshooting page. Each keyword links to a collection of all pages associated with that keyword.                                       |
    |`tags`| array of strings                                     |No| Concepts, actions, or things associated with the troubleshooting entry. These are not displayed in the UI, but they affect the calculation of related pages.                         |
    
    Beneath the front matter, describe the error and its solution in regular Markdown. You can also use any other components allowed within the docs site.
    
    The entry shows up on the troubleshooting pages for its associated products and topics. If the page doesn't already exist, add an entry for it in the page
    index, setting `type` to `placeholder`. See [Navigation tree](#navigation-tree).

> #### Navigation tree

Reviewer: ```suggestion
## Edit the navigation hierarchy for all documentation
```

Not sure about the title

> The navigation hierarchy of a doc section is governed by `page-index/page-index.js` within the corresponding directory. To change the structure, for example, add or delete pages in a section, modify the corresponding `page-index.js`.

Reviewer: Add a line explaining what the structure does. Order and visibility of files etc. 

For example is always at the start of a sentence.

> 

Reviewer: small example please. entry with a child entry and link to where it is in the code.

|-|-|-|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> |`href`|string|Yes| The URL segment to use for the page. If there is a corresponding Markdown file, `href` must match the name of the Markdown file, minus the file extension.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |

Reviewer: Can you use check marks and crosses instead of yes or no. Have a look a the table in https://docs-dev.timescale.com/docs-3452-docs-rfc-update-the-contribution-guide/self-hosted/3452-docs-rfc-update-the-contribution-guide/upgrades/minor-upgrade/#plan-your-upgrade-path.

> |`title`|string|Yes| The title of the page, used as the page name within the TOC on the left.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |

Reviewer: Point out that this MUST be the same as the title in the page metadata and the #headeronetitle

> |`excerpt`|string|No| The short description of the page, used for the page card if `pageComponents` is set to `featured-cards`. See `pageComponents` for details.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |

Reviewer: Point out that this MUST be the same as the excerpt in the page metadata.

Author: It cannot be the same because the limits are different. Content cards can only accommodate around 100 characters, while the SEO description should be up to 200. I will specify the character limit though.

|`featuredChildren`| Array of URLs| No| Similar to `pageComponents`, this displays the children of the current page, but only the selected ones.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|`index`|string|No| If a section landing page needs to be different from the `index.md` file in that directory, this field specifies the corresponding Markdown file name. | 

> #### Partials

Reviewer: ```suggestion
## Reuse text in multiple pages
```

Partials allow you to reuse snippets of content in multiple places. All partials
live in the `_partials` top-level directory. To make a new partial, create a new

> `.md` file in this directory. The filename must start with an underscore. Then import it into the target page and reference in the relevant place. See [Formatting examples](_formatting_examples.md).

Reviewer: Link directly to contributing.md#partials, and give the new title as the link.

### Formatting

In addition to all the [regular Markdown formatting][markdown-syntax], the following elements are available for Timescale docs:

- Procedure blocks 
- Highlight blocks
- Tabs
- Custom code blocks
- Multi-tab code blocks
- Tags

See [Formatting examples](_formatting_examples.md) for how to use them. 

### Variables

> Timescale documentation uses variables for its product names, features, and UI elements in Timescale Console with the following syntax: `$VARIABLE_NAME`. Variables do not work inside the following: 

Reviewer: ```suggestion
Timescale documentation uses variables for its product names, features, and UI elements in  $CONSOLE with the following syntax: `$VARIABLE_NAME`. Variables do not work inside the following: 
```

While I think about it, can use smother this page with variables please.

Author: Even though they will not render on github?

- Front matter on each page
- HTML tables

The main variables include the following:

```text
COMPANY = Timescale
COMPANY_URL = https://www.timescale.com
CLOUD_LONG = Timescale Cloud
TIMESCALE_DB = TimescaleDB
CONSOLE = Timescale Console
CONSOLE_URL = https://console.cloud.timescale.com/
SERVICE_LONG = Timescale Cloud service
SERVICE_SHORT = service
```

The list of variables is an ever-evolving document. Rely on the help of our doc team to apply all relevant ones when reviewing your contribution. 

### Links

> Links should be reference-style links where the link address is at the bottom of the page. 

Reviewer: Can you say reference style in the formatting page please, and link to somewhere that explains what that is. I just think of it as markdown variables in page scope.

Author: Done

- Internal page links: internal links do not need to include the domain name `https://docs.timescale.com`. Use the `:currentVersion:` variable instead of `latest` in the URL.
- External links: input external links as is. 

See [Formatting examples](_formatting_examples.md) for details. 

### Visuals

> When adding screenshots to the docs, aim for a full-screen view to provide better context. Attach the image to your issue or PR, and the doc team will upload and insert it for you.

Reviewer: Full screen, but reduce the size of your browser so there is as little wasted space as possible.

### SEO optimization 

To make a documentation page more visible and clear for Google: 

> - Include a `title` and `excerpt` meta tags at the top of the page. These represent meta title and description required for SEO optimization.

Reviewer: ```suggestion
- Include the `title` and `excerpt` meta tags at the top of the page. These represent the meta title and description required for SEO optimization.
```

>   - `title`: up to 60 characters, a short description of the page contents. In most cases a variation of the page title. 

Reviewer: Add that these must be the same as the TOC js file.

Author: It shouldn't though? This is an SEO title, the one in the TOC makes no difference for SEO and can/should be shorter. Making them the same will either overcrowd the TOC, or under-use the SEO character limit.

  - `excerpt`: under 200 characters, a longer description of the page contents. In most cases a variation of the page intro. 

> - Summarize each paragraph contents in the first sentence of that paragraph. 

Reviewer: ```suggestion
- Summarize the contents of each paragraph in the first sentence of that paragraph. 
```

> - Include main page keywords into the meta tags, page title, first header, and intro. 

Reviewer: point out that this is mostly the features included in the page. Give an example. "Create continuous aggregates in a hypertable ..."
